### PR TITLE
Fix issues around checking for existence of cert in secretstore

### LINKED ISSUE #1675: Implementation of Loading kong cert pair from filesystem to Vault in security-secretstore-setup.
PKI cert pair for KONG API-gateway is created by security-secret-setup and saved into volume. Security-secretstore-setup need to load such cert pair into Vault so that the initialization process of API-Gateway can obtain them from there and apply on the path of API-Gateway.

### LINKED ISSUE #1735: Config-Seed: Populate Global Security Configuration in Registry
Per discussions documented in the Core Working Group 15-Aug-2019, the mechanism by which security features will be enabled will affect security operations for the whole EdgeX platform. In other words, security is not a service by service setting. To that end, we need to have a global configuration area for security in Consul that services are aware of.

The current proposal is to have the following section written by the config-seed

`/edgex/security`

For this particular issue, the above section will contain one key `SECRET_STORE` the value of which will be either true/false. This value will be set by providing the following environment variable to the config-seed -- `edgex_security_secret_store`. This follows the convention already present in the config-seed of overriding values in Consul according to an environment variable that maps to a path.

Separate issues will follow for integrating each service with this global security configuration, as well as usage of environment variables should Consul not be in use.

### PR DESCRIPTION
**Pull request by jdharms**
Fixes #1675  again.

When addressing code review feedback I made a couple of changes that didn't work with the semantics of the rest of the code (there was a case in certs.go:retrieve() where an error was logged and not returned--I "fixed" it to return an error.  This change causes this function to return an error and has the caller handle it appropriately.)

I also remembered why I was checking for any status code in the 2xx range in a function and changed the code to accommodate 204's.

## To test:
1. Inside `cmd/security-secretstore-setup/res/configuration.toml`, edit the `[SecretStore]` section for the following fields:
```
server = "localhost"
scheme = "http"
cafilepath = "/Users/jdharms/workspace/edgex-go/cmd/security-secretstore-setup/EdgeXFoundryCA.pem"
certfilepath = "/Users/jdharms/workspace/edgex-go/cmd/security-secretstore-setup/edgex-kong.pem"
keyfilepath = "/Users/jdharms/workspace/edgex-go/cmd/security-secretstore-setup/edgex-kong.priv.key"
tokenfolderpath = "/Users/jdharms/workspace/edgex-go/cmd/security-secretstore-setup"
```
2. Ensure the paths in the previous step point to files on your file system and not mine.  The contents of these files are not important right now; I just used the filenames.
3. Run vault with flag `-dev-kv-v1` to ensure the right version of the secrets engine (alternatively bring up the v1 kv some other way)
4. Create a resp-init.json file if it doesn't exist inside the tokenfolderpath above.  The contents should be:
```
{
  "keys": [
    "test-keys"
  ],
  "keys_base64": [
    "test-keys-base64"
  ],
  "root_token": "s.MRPe8MVhaSePWcfDdHQsMImy"
}
```
Where the `root_token` value is the root token for your dev mode vault instance.
5. Make build and executing the security-secretstore-setup binary should succeed.

Signed-off-by: Daniel Harms <jdharms@gmail.com>

### CODE REVIEW

**Comment by anonymouse64** [comment]
thanks for the in-depth testing instructions!

**Comment by bnevis-i** [comment]
> thanks for the in-depth testing instructions!

Ditto.  Will come in extremely useful when testing https://github.com/edgexfoundry/edgex-go/issues/1798

**Comment by anonymouse64** [comment]
I missed this in the initial PR adding this to edgex-go so it's not your fault, but the configuration.toml in res/configuration.toml (i.e. _not_ the docker one) should have localhost specified as the host, and sensible default file paths specified so it reasonably works out of the box when run locally in the git tree without the modifications you explained in your testing instructions

It would be nice if you could pick this up in this PR, if not then I will file an issue and someone else can try to pick that up soon

**Comment by jdharms** [comment]
Agree with your nitpicks, will address. I thought I’d added the http thing
to the instructions, and for me they are working with garbage data in the
key files, so not entirely sure what’s going on there. And good catch on
the init flag.

File: internal/security/secretstore/certs.go [hunk 1]
```diff
@@ -95,9 +98,8 @@ func (cs *Certs) retrieve(t string) (*CertPair, error) {
 	cc := CertCollect{}
 
 	if resp.StatusCode == http.StatusNotFound {
-		e := fmt.Errorf("proxy cert pair NOT found in secret store @/%s, status: %s", cs.certPath, resp.Status)
-		LoggingClient.Info(e.Error())
-		return nil, e
+		LoggingClient.Info(fmt.Sprintf("proxy cert pair NOT found in secret store @/%s, status: %s", cs.certPath, resp.Status))
```
[the remark this answers is not in the document]
Comment (jdharms) [reply internal/security/secretstore/certs.go]: Depends on what a logged Error means, I guess.  It's a pretty typical case in the --init flow, so being logged as an error might be misleading.  I think if the caller received no cert pair when it was expecting one then that would be the place to log the error.

[the remark this answers is not in the document]
Comment (anonymouse64) [reply internal/security/secretstore/certs.go]: if that's the case then perhaps this function shouldn't do any logging and the higher level calling function should do the logging as it knows better if failure here is an "error" or not

**Comment by jdharms** [comment]
recheck

File: internal/security/secretstore/certs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdharms) [reply internal/security/secretstore/certs.go]: I like having the info logging here either way--there's nothing wrong with the higher level function logging an error afterwards if necessary.

[the remark this answers is not in the document]
Comment (anonymouse64) [reply internal/security/secretstore/certs.go]: ok

**Comment by anonymouse64** [comment]
any desire in fixing the configuration.toml for the non-docker case in this PR?

**Comment by jdharms** [comment]
> any desire in fixing the configuration.toml for the non-docker case in this PR?

I think I would prefer this to be covered in another PR.

**Comment by tsconn23** [comment]
recheck

**Pull request merged**